redundant-bmc/src/services_impl.cpp

```diff
@@ -331,4 +332,27 @@ bool ServicesImpl::isPoweredOn() const
     throw std::runtime_error("Power state not available");
 }
 
+// NOLINTNEXTLINE
+auto ServicesImpl::getBMCState() const -> sdbusplus::async::task<
+    sdbusplus::common::xyz::openbmc_project::state::BMC::BMCState>
+{
+    using ObjectMapper =
+        sdbusplus::client::xyz::openbmc_project::ObjectMapper<>;
+    using StateMgr = sdbusplus::client::xyz::openbmc_project::state::BMC<>;
+
+    auto mapper = ObjectMapper(ctx)
+                      .service(ObjectMapper::default_service)
+                      .path(ObjectMapper::instance_path);
+
+    std::string statePath = std::string{StateMgr::namespace_path::value} + '/' +
+                            StateMgr::namespace_path::bmc;
+    std::vector<std::string> stateIface{StateMgr::interface};
+
+    auto object = co_await mapper.get_object(statePath, stateIface);
```

Comment:
Suggestion: Would it be beneficial to add a separate helper API for calling the GetObject DBus API? I noticed that the same API is already being called in the readHostState function but, its your wish.

Reply:
done.